examples/language-modeling/main.py

```diff
@@ -196,44 +220,32 @@
                       amp=args.amp, n_samples=args.n_samples, low_gpu_mem_usage=args.low_gpu_mem_usage,
                       seed=args.seed, gradient_accumulate_steps=args.gradient_accumulate_steps,
                       scale_dtype=args.scale_dtype, weight_config=weight_config)  ##TODO args pass
-    model, q_config = autoround.quantize()
+    model, _ = autoround.quantize()
     model_name = args.model_name.rstrip("/")
+    
     export_dir = args.output_dir + "/" + model_name.split('/')[-1] + f"-autoround-w{args.bits}g{args.group_size}"
-
-    if args.deployment_device == 'cpu':
-        export_dir += "-cpu"
-        autoround.save_quantized(output_dir=export_dir)
-        del q_config
-    elif args.deployment_device == 'gpu':
-        export_dir += "-gpu"
-        autoround.save_quantized(export_dir, format="auto_gptq", use_triton=True)
-        model = model.eval()
+    if args.deployment_device == 'gpu' or args.deployment_device == 'both':
+        autoround.save_quantized(f'{export_dir}-gpu', format="auto_gptq", inplace=False, use_triton=True)
+    if args.deployment_device == 'cpu' or args.deployment_device == 'both':
+        autoround.save_quantized(output_dir=f'{export_dir}-cpu', inplace=False)
 
     if args.device != "cpu":
         torch.cuda.empty_cache()
     model.eval()
     output_dir = args.output_dir + "/" + model_name.split('/')[-1] + f"-autoround-w{args.bits}g{args.group_size}-qdq"
 
-    pt_dtype = torch.float16
-    if (hasattr(model, 'config') and (model.dtype is torch.bfloat16 or model.config.torch_dtype is torch.bfloat16)):
-        dtype = 'bfloat16'
-        pt_dtype = torch.bfloat16
-    else:
-        if str(args.device) != "cpu":
-            pt_dtype = torch.float16
-            dtype = 'float16'
-        else:
-            pt_dtype = torch.float32
-            dtype = 'float32'
+    model = model.to(pt_dtype)
+    model = model.to("cpu")
+    model.save_pretrained(output_dir)
+    tokenizer.save_pretrained(output_dir)
 
     excel_name = f"{output_dir}_result.xlsx"
     output_dir += "/"
     print(excel_name, flush=True)
-    if not args.eval_legacy:
-        from eval.evaluation import eval_model
-    else:
-        from eval_legacy.evaluation import eval_model
-    eval_model(output_dir=output_dir, model=model, tokenizer=tokenizer, tasks=args.tasks, \
-               eval_bs=args.eval_bs, use_accelerate=args.low_gpu_mem_usage, device=torch_device, excel_file=excel_name,
-               limit=None)
+    
+    
```

Comment:
add a args to control eval or not, thus we could support tuning large model on one device and eval on multiple device